### PR TITLE
Upgrade Flutter requirement to latest stable(3.13.8)

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
This PR addresses 
- #541 

We have been withholding upgrading Flutter as it requires the huge task of migrating `breezmobile` to null safety. Sticking to 3.7.12 helped us Breez developers to context switch between our Flutter projects, `c-breez`, `sdk-flutter` & `breezmobile` repositories with ease. 

Version management in Flutter is a bit lacking when it comes to this use case but there's a version management CLI tool called fenv, which makes switching between versions much easier. 

We decided forgo this ease in favor of appealing to other Flutter users that use our SDK and use fenv internally.

### Changelist:
- Update Flutter to 3.13.8
  -  Bump Dart range to  '>=3.0.0 <4.0.0'
  -  Bump Flutter range to  '>=3.10.0'
- Update flutter_rust_bridge to 1.80.1 & generate bindings
- Bump sdk-flutter version to 0.3.0

### TODO:
- Detail the changes in CHANGELOG.md

### CODE REVIEW

File: libs/sdk-flutter/pubspec.yaml [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
 description: A new flutter plugin project.
-version: 0.2.5
+version: 0.3.0
 publish_to: none
 
 environment:
-  sdk: ">=2.19.6 <3.0.0"
-  flutter: ">=3.7.12"
+  sdk: '>=3.0.0 <4.0.0'
+  flutter: ">=3.10.0"
```
Comment (ademar111190) [inline libs/sdk-flutter/pubspec.yaml:8]: As we publish a Sdk, keeping the minimum requirement as low as possible is interesting.
So, is it necessary to increase the **minimum** version?

Comment (erdemyerebasmaz) [reply libs/sdk-flutter/pubspec.yaml:8]: Yes, these are the minimum necessary versions.
In Dart 3.0, null safety is built in; you can not turn it off or have a middle-ground, e.g; no `--no-sound-null-safety` flag, and minimum Dart requirement has been upgraded to 3.0 as of Flutter 3.10.0.

Comment (erdemyerebasmaz) [reply libs/sdk-flutter/pubspec.yaml:8]: No, these are the minimum necessary versions due to `flutter_rust_bridge` requirements
`flutter_rust_bridge >=1.76.0 which requires SDK version >=3.0.0 <4.0.0`

which can be downgraded but I think we should move onto Flutter 3.10 & Dart 3.

**Review by ubbabeck — APPROVED** [review]
tACK

File: libs/sdk-flutter/pubspec.yaml (hunk 1 above)
Comment (ademar111190) [reply libs/sdk-flutter/pubspec.yaml:8]: I see; thanks for checking out @erdemyerebasmaz and congrats; that was a hard work

**Review by ademar111190 — APPROVED** [review]

**Review by roeierez — APPROVED** [review]
LGTM

**Comment by ubbabeck** [comment]
If you work on both breezmobile and c-breez  and have to switch between different flutter version i'd like to recommend [direnv](https://github.com/direnv/direnv) as an alternative to fenv.

**Comment by erdemyerebasmaz** [comment]
> If you work on both breezmobile and c-breez and have to switch between different flutter version i'd like to recommend [direnv](https://github.com/direnv/direnv) as an alternative to fenv.

@breez/breez-developers You are free to use any solution you like for Flutter version management. I would only make sure to include any config files they may have to`.gitignore`.

**Comment by ubbabeck** [comment]
> > If you work on both breezmobile and c-breez and have to switch between different flutter version i'd like to recommend [direnv](https://github.com/direnv/direnv) as an alternative to fenv.
> 
> @breez/breez-developers You are free to use any solution you like for Flutter version management. I would only make sure to include any config files they may have to`.gitignore`.

Good point, i'll open a pr for it.

**Pull request merged**